packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1489 @@
+    for (int i = 1; i < parts.length; ++i) {
+      _buffer
+        ..write('\n')
+        ..write(prefixOtherLines)
+        ..write(parts[i]);
+    }
+
+    if (lineTerminated)
+      _buffer.write('\n');
+
+    _atLineStart = lineTerminated;
+  }
+
+  /// Write text assuming the text already obeys the specified prefixes for the
+  /// first and subsequent lines.
+  void writeRaw(String text) {
+    if (text.isEmpty)
+      return;
+    _buffer.write(text);
+    _atLineStart = text.endsWith('\n');
+  }
+
+
+  /// Write a line assuming the line obeys the specified prefixes. Ensures that
+  /// a newline is added if one is not present.
+  /// The same as [writeRaw] except a newline is added at the end of [line] if
+  /// one is not already present.
+  ///
+  /// A new line is not added if the input string already contains a newline.
+  void writeRawLine(String line) {
+    if (line.isEmpty)
+      return;
+    _buffer.write(line);
+    if (!line.endsWith('\n'))
+      _buffer.write('\n');
+    _atLineStart = true;
+  }
+
+  @override
+  String toString() => _buffer.toString();
+}
+
+class _NoDefaultValue {
+  const _NoDefaultValue();
+}
+
+/// Marker object indicating that a DiagnosticNode has no default value.
+const _NoDefaultValue kNoDefaultValue = const _NoDefaultValue();
+
+/// Defines diagnostics data for an [Object].
+///
+/// DiagnosticsNode provides a high quality multi-line string dump via
+/// [toStringDeep]. The core members are the [name], [description], [getProperties],
+/// [object], and [getChildren]. All other members exist typically to provide
+/// hints for how [toStringDeep] and debugging tools should format output.
+abstract class DiagnosticsNode<T> {
+  DiagnosticsNode({
+    @required this.name,
+    this.style: DiagnosticsTreeStyle.sparse,
+    this.showName: true,
+    this.showSeparator: true,
+    this.emptyBodyDescription,
+  }) {
+    // A name ending with ':' indicates that the user forgot that the ':' will
+    // be automatically added for them when generating descriptions of the
+    // property.
+    assert(name == null || !name.endsWith(':'));
+  }
+
+  /// Constructor that creates a [DiagnosticsNode] where properties and children
+  /// are computed lazily.
+  factory DiagnosticsNode.lazy({
+    String name,
+    T object,
+    String description,
+    void fillProperties(List<DiagnosticsNode<Object>> properties),
+    List<DiagnosticsNode<Object>> getChildren(),
+    String emptyBodyDescription,
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.sparse,
+  }) {
+    return new _LazyMembersDiagnosticsNode<T>(
+      name: name,
+      object: object,
+      description: description,
+      fillProperties: fillProperties,
+      getChildren: getChildren,
+      style: style,
+      emptyBodyDescription: emptyBodyDescription,
+    );
+  }
+
+
+  /// Diagnostics containing just a string `message` and not a a concrete name
+  /// or value.
+  static DiagnosticsNode<Object> message(String message, {
```

Comment:
Switched it back to be a factory now that 
DiagnosticsNode does not have a generic argument but DiagnosticsProperty does.
Problem was factory constructors have to have the generic arguments of their class and having users write
`new DiagnosticsNode<Object>.message('foo')`
wouldn't have made sense.